src/explore-education-statistics-frontend/src/middleware/pages/__tests__/redirectPages.test.ts

```diff
@@ -0,0 +1,275 @@
+import _redirectService, {
+  Redirects,
+} from '@frontend/services/redirectService';
+import redirectPages from '@frontend/middleware/pages/redirectPages';
+import { NextResponse, NextRequest } from 'next/server';
+
+jest.mock('@frontend/services/redirectService');
+const redirectService = _redirectService as jest.Mocked<
+  typeof _redirectService
+>;
+
+describe('redirectPages', () => {
+  const redirectSpy = jest.spyOn(NextResponse, 'redirect');
+  const nextSpy = jest.spyOn(NextResponse, 'next');
+
+  const testRedirects: Redirects = {
+    methodologyRedirects: [
+      { fromSlug: 'original-slug-1', toSlug: 'updated-slug-1' },
+      { fromSlug: 'original-slug-2', toSlug: 'updated-slug-2' },
+    ],
+    publicationRedirects: [
+      { fromSlug: 'original-slug-3', toSlug: 'updated-slug-3' },
+      { fromSlug: 'original-slug-4', toSlug: 'updated-slug-4' },
+    ],
+  };
+
+  afterEach(() => {
+    redirectSpy.mockReset();
+  });
+
+  test('does not re-request the list of redirects once it has been fetched', async () => {
+    redirectService.list.mockResolvedValue(testRedirects);
+    const req = new NextRequest(
+      new Request('https://my-env/methodology/original-slug'),
+      {},
```

Comment:
Seems like this arg is optional and can be removed. Same thing for similar cases in this test suite